File: src/main/java/lesson6/MaxProductOfThree.java
package lesson6;

import java.util.Arrays;

public class MaxProductOfThree {
    private int solution(int[] A) {

        if (A.length < 3) {
            return 0;
        }

        Arrays.sort(A);
        int negativeMax = A[0] * A[1] * A[A.length - 1];
        int positiveMax = A[A.length - 1] * A[A.length - 2] * A[A.length - 3];

        return Math.max(negativeMax, positiveMax);
    }

    public static void main(String[] args) {
        MaxProductOfThree m = new MaxProductOfThree();
        int[] A = new int[]{-3, 1, 2, -2, 5, 6};
        System.out.println(m.solution(A));
    }
}

File: src/main/kotlin/lesson6/MaxProductOfThree.kt
package lesson6

private fun solution(A: IntArray): Int {

    if(A.size-3 < 0)
        return -1

    A.sort()
    val negativeMax = A[0] * A[1] * A.last()
    val positiveMax = A.last() * A[A.size-2] * A[A.size-3]

    return negativeMax.coerceAtLeast(positiveMax)
}

fun main() {
    val A = intArrayOf(-3, 1, 2, -2, 5, 6)
    println(solution(A))
}
